### PR TITLE
feat: allow setting local addr on custom spu

### PR DESCRIPTION
**Pull request by nacardin**

### CODE REVIEW

File: crates/fluvio-controlplane-metadata/src/spu/spec.rs [hunk 1]
```diff
@@ -137,6 +137,8 @@ pub struct CustomSpuSpec {
     pub private_endpoint: Endpoint,
     #[cfg_attr(feature = "use_serde", serde(skip_serializing_if = "Option::is_none"))]
     pub rack: Option<String>,
+    #[cfg_attr(feature = "use_serde", serde(skip_serializing_if = "Option::is_none"))]
+    pub public_endpoint_local: Option<Endpoint>,
```
Comment (sehz) [inline crates/fluvio-controlplane-metadata/src/spu/spec.rs:141]: why need to add here?

Comment (nacardin) [reply crates/fluvio-controlplane-metadata/src/spu/spec.rs:141]: This is used in the command to register the custom SPU

Comment (sehz) [reply crates/fluvio-controlplane-metadata/src/spu/spec.rs:141]: It is using CustomSpuSpec,  don't think there is need to add to `SpuSpec`

Comment (nacardin) [reply crates/fluvio-controlplane-metadata/src/spu/spec.rs:141]: `SpuSpec` already has the `public_endpoint_local` field.
This PR adds the field to `CustomSpuSpec`.

Comment (sehz) [reply crates/fluvio-controlplane-metadata/src/spu/spec.rs:141]: Got it. got confused about diff

**Review by sehz — APPROVED** [review]

**Review by EstebanBorai — APPROVED** [review]

**Pull request merged**